src/app/store/Product/Product.reducer.js

```diff
@@ -23,26 +24,33 @@ const initialState = {
 const ProductReducer = (state = initialState, action) => {
     switch (action.type) {
     case UPDATE_PRODUCT_DETAILS:
-        const { product, product: { attributes, variants }, filters } = action;
+        const {
+            product,
+            product: {
+                variants: initialVariants, configurable_options, attributes, type_id
+            }
+        } = action;
+
+        const brand = getBrand(attributes);
 
-        attributes.forEach(({ attribute_code, attribute_value }) => {
-            product[attribute_code] = attribute_value;
-        });
+        const parameters = !initialVariants
+            ? attributes.reduce(
+                (acc, { attribute_code, attribute_value }) => ({ ...acc, [attribute_code]: attribute_value }),
+                {}
+            ) : undefined;
```

Comment:
Please do not do `undefined` to variable that is clearly an empty object. Write this reduce like this:

```js
attributes.reduce((acc, { attribute_code, attribute_value }) => (
    { ...acc, [attribute_code]: attribute_value }
), {});
```